dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -454,68 +454,136 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+        Contentlet contentletSecondLang = contentletAPI
+                .find(contentletDefaultLang.getInode(), systemUser, false);
+        contentletSecondLang.setInode("");
+        contentletSecondLang.setLanguageId(secondLang.getId());
+        contentletSecondLang.setStringProperty("firstName", firstName);
+        contentletSecondLang = contentletAPI.checkin(contentletSecondLang, systemUser, false);
+        ContentletDataGen.publish(contentletSecondLang);
+    }
+
+    private Contentlet createAndPublishEmployeeContent(Host site, Language language,
+            String firstName) throws DotDataException, DotSecurityException {
+        Contentlet contentletDefaultLang = TestDataUtils
+                .getEmployeeContent(true, language.getId(), null, site);
+
+        contentletDefaultLang = contentletAPI
+                .find(contentletDefaultLang.getInode(), systemUser, false);
+        contentletDefaultLang.setStringProperty("firstName", firstName);
+        contentletDefaultLang = contentletAPI.checkin(contentletDefaultLang, systemUser, false);
+        ContentletDataGen.publish(contentletDefaultLang);
+        return contentletDefaultLang;
+    }
+
+
+    /**
+     * Given sceneario: Content in second language only referenced from a page in default language, and having DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true.
+     * Create a Site-Search run including the second language only.
+     * Expected result: the page should be in the results, because DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true and DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=true .
+     */
+
+    @Test
+    public void test_ContentInSecondLang_PageInDefaultLang_IndexingSecondLang_ShouldFindResults()
+            throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
+
+        boolean defaultContentToDefaultLangOriginalValue =
+                Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+
+        try {
+
+            Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", true);
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
+            folder = new FolderDataGen().site(site).nextPersisted();
 
-            HTMLPageDataGen.publish(page);
+            Contentlet contentSecondLang = createAndPublishEmployeeContent(site, secondLang, "catalina");
 
-            final MultiTree multiTree = new MultiTree(page.getIdentifier(),
-                    container.getIdentifier(),
-                    contentletLang1.getIdentifier(), getDotParserContainerUUID(uuid), 0);
+            // create container, template and htmlpage to reference previously created content
 
-            APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+            HTMLPageAsset page = createHtmlPageAsset(defaultLang, contentSecondLang);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'page' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)